### PR TITLE
Update to Aspire preview.4

### LINKED ISSUE #225: [WebToolsE2E][Aspire] After updating Aspire 8.0 P4, opening eShop.Web solution fails with error:"NpgsqlEntityFrameworkCorePostgreSQLSettings' does not contain a definition for 'DbContextPooling' and no accessible extension method 'DbContextPooling' accepting a first argument of type 'NpgsqlEntityFrameworkCorePostgreSQLSettings' could be found".
**REGRESSION INFO:** Worked fine on Aspire 8.0.0-preview.3.24105.21

**INSTALL STEPS**

1. Clean machine: Win11 x64 23h2 ENU
2. Install VS 17.10 Preview 2 [ Includes Aspire 8.0.0-preview.4.24129.7 ]
3. Apply NuGet dotnet-tools Feeds
4. Install Docker Desktop for Windows

**REPRO STEPS**

1. Clone eShop repo from main branch ( [Bug 1985774](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems/edit/1985774) will be encountered after cloning)
2. Update nuget.config to use dotnet-tools feed
- image
3. Update Aspire version in Directory.Packages.props
- image
4. Open eShop.Web.slnf and build solution

**ACTUAL:**
Error	CS1061	'IDistributedApplicationBuilder' does not contain a definition for 'AddPostgresContainer' and no accessible extension method 'AddPostgresContainer' accepting a first argument of type 'IDistributedApplicationBuilder' could be found (are you missing a using directive or an assembly reference?)
Error	CS0246	The type or namespace name 'PostgresContainerResource' could not be found (are you missing a using directive or an assembly reference?)
Error	CS1061	'NpgsqlEntityFrameworkCorePostgreSQLSettings' does not contain a definition for 'DbContextPooling' and no accessible extension method 'DbContextPooling' accepting a first argument of type 'NpgsqlEntityFrameworkCorePostgreSQLSettings' could be found (are you missing a using directive or an assembly reference?)	
Error	CS0006	Metadata file 'C:\Users\v-sherryfan\source\repos\eShop\src\Ordering.API\obj\Debug\net8.0\ref\Ordering.API.dll' could not be found
Error	CS1061	'IDistributedApplicationBuilder' does not contain a definition for 'AddRedisContainer' and no accessible extension method 'AddRedisContainer' accepting a first argument of type 'IDistributedApplicationBuilder' could be found (are you missing a using directive or an assembly reference?)
Error	CS0006	Metadata file 'C:\Users\v-sherryfan\source\repos\eShop\src\Ordering.API\obj\Debug\net8.0\ref\Ordering.API.dll' could not be found
Error	CS1061	'IDistributedApplicationBuilder' does not contain a definition for 'AddRabbitMQContainer' and no accessible extension method 'AddRabbitMQContainer' accepting a first argument of type 'IDistributedApplicationBuilder' could be found (are you missing a using directive or an assembly reference?)
image

**Additional Info**
image

**EXPECTED**
Opening solution without any errors.

### PR DESCRIPTION
**Pull request by DamianEdwards**
Can't merge until preview.4 is released at which point the change to `nuget.config` must be reverted.

Fixes #225

### CODE REVIEW

**Comment by DamianEdwards** [comment]
@adityamandaleeka this is unblocked now that preview.4 is released. The package version needs to be updated to the released version and the nuget.config changes need to be reverted.

File: Directory.Packages.props [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <AspnetVersion>8.0.2</AspnetVersion>
     <MicrosoftExtensionsVersion>8.2.0</MicrosoftExtensionsVersion>
     <EfVersion>8.0.2</EfVersion>
-    <AspireVersion>8.0.0-preview.3.24105.21</AspireVersion>
+    <AspireVersion>8.0.0-preview.4.24129.7</AspireVersion>
```
Comment (eerhardt) [inline Directory.Packages.props:8]:
```suggestion
    <AspireVersion>8.0.0-preview.4.24156.9</AspireVersion>
```

File: nuget.config [hunk 2]
```diff
@@ -3,11 +3,16 @@
   <packageSources>
     <!--To inherit the global NuGet package sources remove the <clear/> line below -->
     <clear />
+    <add key="dotnet-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" protocolVersion="3" />
```
Comment (eerhardt) [inline nuget.config:6]:
```suggestion
```

File: nuget.config [hunk 3]
```diff
@@ -3,11 +3,16 @@
...
     <add key="nuget" value="https://api.nuget.org/v3/index.json" />
   </packageSources>
 
   <packageSourceMapping>
     <!-- key value for <packageSource> should match key values from <packageSources> element -->
+    <clear />
+    <packageSource key="dotnet-tools">
+      <package pattern="*" />
+    </packageSource>
```
Comment (eerhardt) [inline nuget.config:15]:
```suggestion
```

**Comment by eerhardt** [comment]
> The package version needs to be updated to the released version and the nuget.config changes need to be reverted.

Done.

**Review by adityamandaleeka — APPROVED** [review]

**Pull request merged**